fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    async function read() {
+      try {
+        const url = `https://jsonplaceholder.typicode.com/users/${userId}`;
+        const response = await fetch(url);
+        if (!response) throw new Error(`cannot fetch ${url}`);
```

Comment:
`response` will always be defined so this check isn't quite right. You should be checking `response.ok`.